spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -235,6 +199,37 @@ Map<StructLike, List<List<FileScanTask>>> planFileGroups(long startingSnapshotId
     }
   }
 
+  private StructLikeMap<List<FileScanTask>> groupByPartition(
+      StructType partitionType, Iterable<FileScanTask> tasks) {
+    StructLikeMap<List<FileScanTask>> filesByPartition = StructLikeMap.create(partitionType);
```

Comment:
This should continue to use `table.spec()` and we don't need `coercePartition`. The old behavior of grouping all files from an olde spec should be preserved.

Reply:
Thats strange. So, we don't have enough testcases to catch this behaviour? 

I have reverted it back now. Also, I didn't use `transform` in the end of `planFileGroups` because it needs a special check of `fileGroups.size() > 0` which is not handled in transform